File: JavaScript/isPrime.js
/*
|--------------------------------------------------------------------------
| Is a number prime?
|--------------------------------------------------------------------------
|
*/

function isPrime(num) {
   for (let i = 2, s = Math.sqrt(num); i <= s; i++)
      if (num % i === 0) return false;
   return num > 1;
}


function isPrime(num) {
   for (let i = 2; i <= num ** .5; i++) {
      if (!(num % i)) return false;
   }
   return num > 1;
}


function isPrime(num) {
   let even = 2;
   while ((even * even <= num) && (num % even != 0)) {
      even++;
   }
   return even * even > num && num > 1
}


function isPrime(num) {
   for (let i = 2; i <= num / (i - 1); i++) {
      if (num !== i && num % i === 0) return false;
   }
   return (num > 1)
}


console.log(isPrime(0), false, "0 is not prime");
console.log(isPrime(1), false, "1 is not prime");
console.log(isPrime(2), true, "2 is prime");
console.log(isPrime(73), true, "73 is prime");
console.log(isPrime(75), false, "75 is not prime");
console.log(isPrime(-1), false, "-1 is not prime");

File: Python/6py 1/is_prime.py
"""
|--------------------------------------------------------------------------
| Is a number prime?
|--------------------------------------------------------------------------
|
"""

def is_prime(num):
   even = 2
   while ((even * even <= num) and (num % even != 0)):
      even += 1
   return even * even > num and num > 1


from math import sqrt
def is_prime(n):
   return n > 1 and all(n % d for d in xrange(2, int(sqrt(n)) + 1))


def is_prime(n):
   return n>1 and all(n%i for i in range(2, int(n**.5+1)))


import gmpy2
def is_prime(num):
   return gmpy2.is_prime(num) if num > 0 else False


print(is_prime(0),  False)
print(is_prime(1),  False)
print(is_prime(2),  True)
print(is_prime(73), True)
print(is_prime(75), False)
print(is_prime(-1), False)